contracts/cw1155-base/src/contract.rs

```diff
@@ -0,0 +1,1239 @@
+    IsApprovedForAllResponse, TokenId, TokenInfoResponse, TokensResponse, TransferEvent,
+};
+use cw2::set_contract_version;
+
+use crate::error::ContractError;
+use crate::msg::InitMsg;
+use crate::state::{APPROVES, BALANCES, MINTER, TOKENS};
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:cw1155-base";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+const DEFAULT_LIMIT: u32 = 10;
+const MAX_LIMIT: u32 = 30;
+
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InitMsg,
+) -> StdResult<Response> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+    let minter = deps.api.canonical_address(&msg.minter)?;
+    MINTER.save(deps.storage, &minter)?;
+    Ok(Response::default())
+}
+
+/// To mitigate clippy::too_many_arguments warning
+pub struct ExecuteEnv<'a> {
+    deps: DepsMut<'a>,
+    env: Env,
+    info: MessageInfo,
+}
+
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: Cw1155ExecuteMsg,
+) -> Result<Response, ContractError> {
+    let env = ExecuteEnv { deps, env, info };
+    match msg {
+        Cw1155ExecuteMsg::SendFrom {
+            from,
+            to,
+            token_id,
+            value,
+            msg,
+        } => execute_send_from(env, from, to, token_id, value, msg),
+        Cw1155ExecuteMsg::BatchSendFrom {
+            from,
+            to,
+            batch,
+            msg,
+        } => execute_batch_send_from(env, from, to, batch, msg),
+        Cw1155ExecuteMsg::Mint {
+            to,
+            token_id,
+            value,
+            msg,
+        } => execute_mint(env, to, token_id, value, msg),
+        Cw1155ExecuteMsg::BatchMint { to, batch, msg } => execute_batch_mint(env, to, batch, msg),
+        Cw1155ExecuteMsg::Burn {
+            from,
+            token_id,
+            value,
+        } => execute_burn(env, from, token_id, value),
+        Cw1155ExecuteMsg::BatchBurn { from, batch } => execute_batch_burn(env, from, batch),
+        Cw1155ExecuteMsg::ApproveAll { operator, expires } => {
+            execute_approve_all(env, operator, expires)
+        }
+        Cw1155ExecuteMsg::RevokeAll { operator } => execute_revoke_all(env, operator),
+    }
+}
+
+/// When from is None: mint new coins
+/// When to is None: burn coins
+/// When both are None: not token balance is changed, meaningless but valid
+fn execute_transfer_inner<'a>(
+    deps: &'a mut DepsMut,
+    from: Option<&'a HumanAddr>,
+    to: Option<&'a HumanAddr>,
+    token_id: &'a str,
+    amount: Uint128,
+) -> Result<TransferEvent<'a>, ContractError> {
+    if let Some(from) = from {
+        let from_raw = deps.api.canonical_address(from)?;
+        BALANCES.update(
+            deps.storage,
+            (from_raw.as_slice(), token_id),
+            |balance: Option<Uint128>| balance.unwrap_or_default() - amount,
```

Comment:
Hmmm... do you want `checked_add` here? 
Or are you waiting for the next cosmwasm-std release?

Reply:
I think we can change them at once in the cosmwasm-std upgrade.